addons/purchase_stock/models/purchase.py

```diff
@@ -311,9 +311,10 @@ def _compute_qty_received(self):
                                 [("id", "child_of", move.warehouse_id.view_location_id.id)]
                             )
                         ):
-                            total -= move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom, rounding_method='HALF-UP')
+                            total -= move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom, round=False)
                         else:
-                            total += move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom, rounding_method='HALF-UP')
+                            total += move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom, round=False)
+                total = float_round(total, precision_rounding=line.product_uom.rounding, rounding_method='DOWN')
```

Comment:
Why using DOWN? We use HALF-UP everywhere

Reply:
To never pretend to have delivered more than the reality

Reply:
I'm on the same side than you on this but I already had this discussion with a lot of people and suggesting something closer to reality is often what they prefer. The opposite view, often it's a detail and with a down you always set the lose on the vendor side instead of having something fair between vendor and customer

Reply:
Ok ok I didn't know about this point of view
I changed the rounding method to `HALF-UP` and kept the idea to sum all quantities before the rounding, what do you think ?
It does not fix all cases: for instance in the above case, if the user doesn't create the backorder, he will have a received quantity equal to 2, but at least this value is not more than expected (on step 4) ?